gapis/api/vulkan/transform_make_attachment_readable.go

```diff
@@ -0,0 +1,320 @@
+// Copyright (C) 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vulkan
+
+import (
+	"context"
+
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/memory"
+)
+
+type makeAttachmentReadable2 struct {
+	imagesOnly  bool
+	allocations *allocationTracker
+}
+
+func newMakeAttachmentReadable2(imagesOnly bool) *makeAttachmentReadable2 {
+	return &makeAttachmentReadable2{
+		imagesOnly:  imagesOnly,
+		allocations: nil,
+	}
+}
+
+func (attachmentTransform *makeAttachmentReadable2) RequiresAccurateState() bool {
+	return false
+}
+
+func (attachmentTransform *makeAttachmentReadable2) BeginTransform(ctx context.Context, inputCommands []api.Cmd, inputState *api.GlobalState) ([]api.Cmd, error) {
+	attachmentTransform.allocations = NewAllocationTracker(inputState)
+	return inputCommands, nil
+}
+
+func (attachmentTransform *makeAttachmentReadable2) EndTransform(ctx context.Context, inputCommands []api.Cmd, inputState *api.GlobalState) ([]api.Cmd, error) {
+	return inputCommands, nil
+}
+
+func (attachmentTransform *makeAttachmentReadable2) ClearTransformResources(ctx context.Context) {
+	// Melih TODO: This transform seems to be never releasing the allocations
+	// Check if it's intended
+	// attachmentTransform.allocations.FreeAllocations()
```

Comment:
Most of your TODOs are likely to get solved by subsequent PRs related to the transform refactor, but some others like this one may not. It would be best to track them with proper bugs and `TODO(b/123456)` in the code.